tests/components/aosmith/test_device.py

```diff
@@ -0,0 +1,24 @@
+"""Tests for the device created by the A. O. Smith integration."""
+
+from homeassistant.components.aosmith.const import DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+
+from tests.common import MockConfigEntry
+
+
+async def test_device_nickname(
+    hass: HomeAssistant, init_integration: MockConfigEntry
+) -> None:
+    """Test creation of the device."""
+    device_registry = dr.async_get(hass)
```

Comment:
Device registry is available as test fixture, let me know if you can't find them as I'm on the public transport right now

Reply:
Fixed.